force-app/main/default/classes/SOQL.cls

```diff
@@ -314,18 +409,36 @@ public with sharing class SOQL {
      */
     public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
         // Query the original Opportunity Line Items
-
+        List<OpportunityLineItem> origLineItems = [
+            SELECT PricebookEntryId, Quantity, UnitPrice, TotalPrice, ServiceDate, Description
+            FROM   OpportunityLineItem
+            WHERE  OpportunityId = :origOppId
+        ];
         // Create a list to hold the new line items
+        List<OpportunityLineItem> clonedLineItems = new List<OpportunityLineItem>();
 
         // Generate a random number between 1 and 3
+        
 
         // Loop through each cloned Opportunity Id
-
-            // Loop through each original line item
-
-                // Clone the original line item and add the random number to the quantity
-
-        // Insert the new line items
+        for (Id cloneOppId : cloneOppIds) {
+            for (OpportunityLineItem origItem : origLineItems) {
+                // Clone the original line item and add a random number between 1 and 3 to the quantity
+                OpportunityLineItem clonedItem = new OpportunityLineItem();
+                clonedItem.OpportunityId = cloneOppId;
+                clonedItem.PricebookEntryId = origItem.PricebookEntryId;
+                clonedItem.Quantity = origItem.Quantity + ((Integer) (Math.random() * 3)) + 1; // Adds a random number between 1 and 3
+                clonedItem.UnitPrice = origItem.UnitPrice; // Set UnitPrice only
+                // Dont set TotalPrice here, will be calculated based on UnitPrice * Quantity
+                clonedItem.ServiceDate = origItem.ServiceDate;
+                clonedItem.Description = origItem.Description;
```

Comment:
You can also use .clone() method for SObjects to create a copy. Shortcut if you have a lot of fields to copy over.